loadbalancer/zk/client.go

```diff
@@ -0,0 +1,82 @@
+package zk
+
+import (
+	"github.com/go-kit/kit/log"
+	"github.com/samuel/go-zookeeper/zk"
+	"strings"
+	"time"
+)
+
+const ()
+
+// Client is a wrapper arround the zookeeper client.
+type Client interface {
+	// GetEntries queries the path in zookeeper, places a watch on it
+	// and retrieves data from its current child nodes
+	GetEntries(path string) ([]string, <-chan zk.Event, error)
+	// createParentNodes tries to create the path
+	// in case it does not exist yet on Zookeeper
+	createParentNodes(path string) error
+}
+
+type client struct {
+	*zk.Conn
+	evtConn <-chan zk.Event
+	acl     []zk.ACL
+	quit    chan struct{}
+}
+
+// NewClient returns a *zk.Client with a connection to the named server cluster.
+// It will return an error if the server cluster cannot be resolved.
+func NewClient(servers []string, acl *[]zk.ACL, logger log.Logger) (Client, error) {
+
+	conn, evt, err := zk.Connect(servers, 5*time.Second, withLogger(logger))
+	if err != nil {
+		return nil, err
+	}
+
+	if acl == nil {
+		// no ACL was given, use most permissive ACL possible
+		defaultAcl := zk.WorldACL(zk.PermAll)
+		acl = &defaultAcl
+	}
```

Comment:
I think there's no reason to allow this parameter to be optional. Clients should be able to pass `zk.WorldACL(zk.PermAll)` if they don't care about permissions.